playbooks/roles/request-header-identity-provider/README.md

```diff
@@ -0,0 +1,63 @@
+Request header identity provider test
+=====================================
+
+This ansible role will configure request header as an identity provider to authenticate users with cluster OAUTH server and grants *"cluster-reader"* as a default user role permission.
+
+
+
+Requirements
+------------
+ - Running OCP 4.x cluster is needed.
+ - Access to the cluster with the cluster-admin role. 
+
+Dependencies
+------------
+ - None
+
+
+Role Variables
+--------------
+
+| Variable                          | Required | Default                  | Comments                                                                                                                                                                                                                                                            |
+|-----------------------------------|----------|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| request_header_provider           | yes      | false                    | Set to true to enable request header test execution                                                                                                                                                                                                                 |
```

Comment:
Since default is given why is it Required? Variables are Required only when no default is given and is used somewhere when the flag is enabled.

Reply:
The flag has been set with default `false` and mentioned as Required because it requires to be set to `true` and passed as an external argument by user to enable the role execution.

Reply:
If I don't need this feature I don't have to touch the flag at all. Why is it required then?

Reply:
>Variables are Required only when no default is given and is used somewhere when the flag is enabled.

Reply:
> If I don't need this feature I don't have to touch the flag at all. Why is it required then?

Yes it is correct, unless the role is to be enabled the flag doesn't need to be changed and shouldn't be Required too.  The above comment has cleared the confusion about the Required variable and variable with the defaults, Thank you !

Reply:
> If I don't need this feature I don't have to touch the flag at all. Why is it required then?

Marked the flag variable as not Required in the Roles variables as the flag has default.